libs/ml/tests/ml/model/dnn_regressor_test.cpp

```diff
@@ -88,24 +90,24 @@ bool RunTest(fetch::ml::optimisers::OptimiserType optimiser_type,
   ModelType model = SetupModel<TypeParam, DataType, ModelType>(optimiser_type, model_config,
                                                                train_data, train_labels);
 
+  DataType loss{0};
+  model.Train(1, loss);
+
   // test loss decreases
   fetch::math::SizeType count{0};
   while (count < n_training_steps)
   {
-    DataType loss{0};
     DataType later_loss{0};
```

Comment:
Are we trying to check here that loss decreases every training step? If so then loss and later_loss should be defined outside the loop, and after doing one training step compare the values then set loss = later_loss.

Reply:
I had some issues with Adam and Momentum, where a loss could jump up.